crates/autopilot/src/on_settlement_event_updater.rs

```diff
@@ -107,11 +107,13 @@ impl OnSettlementEventUpdater {
         let hash = H256(event.tx_hash.0);
         tracing::debug!("updating settlement details for tx {hash:?}");
 
-        let transaction = self
-            .eth
-            .transaction(hash)
-            .await?
-            .with_context(|| format!("no tx {hash:?}"))?;
+        let transaction = match self.eth.transaction(hash).await? {
+            Some(tx) => tx,
+            None => {
+                tracing::warn!(?hash, "no tx found, reorg happened");
+                return Ok(false);
+            }
+        };
```

Comment:
nit: slightly more concise with
```suggestion
        let Some(transaction) = self.eth.transaction(hash).await? else {
            tracing::warn!(?hash, "no tx found, reorg happened");
            return Ok(false);
        };
```

Reply:
👍